samples/python/multiview_calibration.py

```diff
@@ -0,0 +1,220 @@
+    print('rvecs', Rs)
+    print('tvecs', Ts)
+    print('K', Ks)
+    print('distortion', distortions)
+    print('mean RMS error over all visible frames %.3E' % errors_per_frame[errors_per_frame > 0].mean())
+
+    visibility_idxs = np.stack(np.where(visibility)) # 2 x M, first row is camera idx, second is frame idx
+    plotCamerasPosition(Rs, Ts, image_sizes)
+    def plot(cam_idx, frame_idx):
+        image = None if image_names is None else cv.cvtColor(cv.imread(image_names[cam_idx][frame_idx]), cv.COLOR_BGR2RGB)
+        plotProjection(image_points[cam_idx][frame_idx], pattern_points, rvecs0[frame_idx],
+            tvecs0[frame_idx], rvecs[cam_idx], Ts[cam_idx], Ks[cam_idx], distortions[cam_idx],
+            is_fisheye[cam_idx], image_sizes[cam_idx], image)
+    plot(visibility_idxs[0,0], visibility_idxs[1,0])
+    plt.show()
+
+def chessboard_points(grid_size, dist_m):
+    pattern = np.zeros((grid_size[0]*grid_size[1],3), np.float32)
+    pattern[:,:2] = np.mgrid[0:grid_size[0],0:grid_size[1]].T.reshape(-1,2)*dist_m # only for (x,y,z=0)
+    return pattern
+
+def circles_grid_points(grid_size, dist_m):
+    pattern = []
+    for i in range(grid_size[0]):
+        for j in range(grid_size[1]):
+            pattern.append([j*dist_m, i*dist_m, 0]);
+    return np.array(pattern, dtype=np.float32)
+
+def asym_circles_grid_points(grid_size, dist_m):
+    pattern = []
+    for i in range(grid_size[0]):
+        for j in range(grid_size[1]):
+            pattern.append([(2*j + i % 2)*dist_m, i*dist_m, 0]);
+    return np.array(pattern, dtype=np.float32)
+
+def calibrateFromImages(files_with_images, grid_size, pattern_type, is_fisheye, dist_m, RESIZE_IMAGE=True):
+    """
+    files_with_images: NUM_CAMERAS x NUM_FRAMES x string - path to image file
+    grid_size: [width, height] -- size of grid pattern
+    dist_m: length of a grid cell
+    is_fisheye: NUM_CAMERAS (bool)
+    """
+    if pattern_type.lower() == 'checkerboard':
+        pattern = chessboard_points(grid_size, dist_m)
+    elif pattern_type.lower() == 'circles':
+        pattern = circles_grid_points(grid_size, dist_m)
+    elif pattern_type.lower() == 'acircles':
+        pattern = asym_circles_grid_points(grid_size, dist_m)
+    else:
+        raise "Pattern type is not implemented!"
+
+    assert len(files_with_images) == len(is_fisheye) and len(grid_size) == 2
+    image_points_cameras = []
+    image_sizes = []
+    criteria = (cv.TERM_CRITERIA_EPS + cv.TERM_CRITERIA_MAX_ITER, 50, 0.001)
+    all_images_names = []
+    for filename in files_with_images:
+        images_names = open(filename, 'r').readlines()
+        for i in range(len(images_names)):
+            images_names[i] = images_names[i].replace('\n', '')
+        all_images_names.append(images_names)
+        image_points_camera = []
+        img_size = None
+        for img_name in images_names:
+            print(img_name)
+            assert os.path.exists(img_name)
```

Comment:
I propose to skip empty lines, if any. It allows to not store frames, where the pattern is not visible.

Reply:
added in new commit: https://github.com/opencv/opencv/pull/22363/commits/af8359e79347c6d6a5f57e2ead5d3f340fc5fa4e